src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -35,27 +36,32 @@ static int get_xclbin_iplayout(char *buffer, XmaXclbinInfo *xclbin_info);
 static int get_xclbin_mem_topology(char *buffer, XmaXclbinInfo *xclbin_info);
 static int get_xclbin_connectivity(char *buffer, XmaXclbinInfo *xclbin_info);
 
-char *xma_xclbin_file_open(const char *xclbin_name)
+std::vector<char> xma_xclbin_file_open(const char *xclbin_name)
 {
     xma_logmsg(XMA_INFO_LOG, XMAAPI_MOD, "Loading %s\n", xclbin_name);
 
-    std::ifstream file(xclbin_name, std::ios::binary | std::ios::ate);
-    std::streamsize size = file.tellg();
-    file.seekg(0, std::ios::beg);
+    std::ifstream infile(xclbin_name, std::ios::binary | std::ios::ate);
+    std::streamsize xclbin_size = infile.tellg();
+    infile.seekg(0, std::ios::beg);
 
-    char *buffer = (char*)malloc(size);
-    if (buffer == NULL) {
+    std::vector<char> xclbin_buffer;
+    try {
+        xclbin_buffer.reserve(xclbin_size);
+    } catch (const std::bad_alloc& ex) {
         xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Could not allocate buffer for file %s\n", xclbin_name);
-        return NULL;
+        xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Buffer allocation error: %s\n", ex.what());
+        throw;
+    } catch (...) {
+        xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Could not allocate buffer for xclbin file %s\n", xclbin_name);
+        throw;
```

Comment:
drop the try/catch, just let the exception peculate up the stack; these log messages are not very interesting and only complicates the code

Reply:
keeping it to capture alloc exception error in log. XMA tries to capture all errors in log. XRT verbosity is used by users to control what they want to see.

Reply:
You are rethrowing, so what is the caller doing with the exception?  Also catch (...) is not needed since std::vector throws std::exception only.

Reply:
there can be length_error as well so good to have catch(...). It is a fatal error so execution can stop. xma log does need log msg to say that loading of xclbin lead to error